src/main/java/com/targus/ui/Mediator.java

```diff
@@ -0,0 +1,113 @@
+package com.targus.ui;
+
+import com.targus.base.OptimizationProblem;
+import com.targus.base.Solution;
+import com.targus.ui.controllers.EmptyController;
+import com.targus.ui.controllers.InformativeController;
+import com.targus.ui.controllers.InputsController;
+import com.targus.ui.controllers.MapController;
+import com.targus.ui.widgets.PotentialPosition;
+import com.targus.ui.widgets.Sensor;
+import com.targus.ui.widgets.Target;
+import javafx.beans.property.ReadOnlyDoubleProperty;
+
+public class Mediator {
+    private InputsController inputsController;
+    private InformativeController informativeController;
+    private MapController mapController;
+    private EmptyController emptyController;
+
+    public void setInputsController(InputsController inputsController) {
+        this.inputsController = inputsController;
+    }
+
+    public void setInformativeController(InformativeController informativeController) {
+        this.informativeController = informativeController;
+    }
+
+    public void setMapController(MapController mapController) {
+        this.mapController = mapController;
+    }
+
+    public void setEmptyController(EmptyController emptyController) {
+        this.emptyController = emptyController;
+    }
+
+    public void resizeMapPane(double width, double height) {
+        mapController.resizePane(width, height);
+    }
+
+    public void display() {
+        informativeController.display();
+    }
+
+    public void addChild(Object child) {
```

Comment:
In naming those communication methods, I have taken initiative and named them as concisely as possible. But you are right, I will make them more descriptive.

Reply:
Apart from making the names more descriptive, we can make them more generic by using interfaces. We can have a meeting to talk about it since I do not know much about the implementation details.

Reply:
For now, I am planning to keep them as is. I will design an interface that would address your concerns. But for this pull request I think it is okay if I leave them like that. Please can you ignore this one temporarily for this pull request?

Reply:
Sure